### PR TITLE
[GraphQL Plugin] Extract plugin related logic

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
Dependent on other PRs. Merging should be synchronised.

File: server/handlers/extensions.go
Comment (leecalcote) [inline server/handlers/extensions.go:1]: In this context, what is an "extension"?

File: server/extensions/input.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package extensions
+
+import (
+	"github.com/layer5io/meshery/server/machines"
+	"github.com/layer5io/meshkit/broker"
+	"github.com/layer5io/meshkit/database"
+	"github.com/layer5io/meshkit/logger"
+)
+
+type ExtensionInput struct {
```
Comment (leecalcote) [inline server/extensions/input.go:10]: Extension, huh?

Comment (MUzairS15) [reply server/extensions/input.go:10]: The PR extracts the constructs that are being used for the go plugin from `models` into its own package.

**Pull request merged**